Gems/LyShine/Code/Source/UiTextComponent.cpp

```diff
@@ -3381,6 +3381,15 @@ void UiTextComponent::Activate()
     FontNotificationBus::Handler::BusConnect();
     LanguageChangeNotificationBus::Handler::BusConnect();
 
+#if defined(CARBONATED)
+    // Reload the font families if they were deleted while this UiTextComponent was inactive.
+    // It is important for UiTextComponent which can be activated and deactivated many times.
+    if (m_fontFamily->familyName.empty())
+    {
+        OnFontsReloaded();
+    }
+#endif
```

Comment:
I'm not a fan of this change - OnFontsReloaded is a bus event that is called by FontNotificationBus.  Manually triggering it here will cause confusion.

Could we call "ChangeFont" in ::Init instead?  Instead of trying to reload the fonts in Activiate?

Reply:
A) "OnFontsReloaded" has been replaced to "ChangeFont"
B) "ChangeFont" is called in Init() but it does not help because Init() is called once, but Activate()/Deactivate() can be called bunch of times. For explanation - see screenshot of crash stack, Activate() is here, Init()
 was not called. ![Init_is_not_called_on_the_each_activation_of_UiTextComponent](https://github.com/user-attachments/assets/0065d77d-8976-4a91-bdbc-38f8a1ffabbc)